### PR TITLE
WT-8484 Improving the performance of the failpoint code

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: src/include/misc_inline.h [hunk 1]
```diff
@@ -221,6 +221,34 @@ __wt_timing_stress(WT_SESSION_IMPL *session, u_int flag)
...
+
+    /* Assert that the given probability is sane. */
+    WT_ASSERT(session, probability >= 0 && probability <= 100);
+
+    /* To support two decimal places we multiply the percent change of occurring by 100. */
+    ratio = (uint32_t)(probability * 100);
+
+    if (__wt_random(&session->rnd) % 10000 <= ratio)
+        return (true);
+    return (false);
```
Comment (keithbostic) [inline src/include/misc_inline.h:249]: Why not just `return = (__wt_random() <= ratio);`

File: src/include/misc_inline.h [hunk 2]
```diff
@@ -221,6 +221,34 @@ __wt_timing_stress(WT_SESSION_IMPL *session, u_int flag)
...
 
+/*
+ * __wt_failpoint --
+ *     A generic failpoint function, it will return true if the failpoint triggers. Takes a double
+ *     representing the probability of the failpoint occurring. Supports percentages with two
+ *     decimal places.
+ */
+static inline bool
+__wt_failpoint(WT_SESSION_IMPL *session, uint64_t conn_flag, double probability)
+{
```
Comment (keithbostic) [inline src/include/misc_inline.h:232]: This doesn't need a double, does it?

**Review by keithbostic — APPROVED** [review]
lgtm

File: src/include/misc_inline.h (hunk 2 above)
Comment (luke-pearson) [reply src/include/misc_inline.h:232]: Nah I just feel like double is more human readable but I will update it to not be an improve the comment so its clear what the variable is.

Comment (luke-pearson) [reply src/include/misc_inline.h:231]: Fixed in 8801a29

File: src/include/misc_inline.h (hunk 1 above)
Comment (luke-pearson) [reply src/include/misc_inline.h]: Fixed in 8801a29

**Review by keithbostic — APPROVED** [review]
lgtm, modulo compiler failures.

**Pull request merged**